climada/util/interpolation.py

```diff
@@ -186,3 +191,44 @@ def index_nn_haversine(centroids, coordinates, threshold=THRESHOLD):
 
     # Copy result to all exposures and return value
     return np.squeeze(assigned[inv])
+
+
+def index_nn_euclidian(centroids, coordinates, threshold=THRESHOLD):
+    """Compute the neareast centroid for each coordinate using a k-d tree.
+
+    Parameters
+    ----------
+    centroids : 2d array
+        First column contains latitude, second column contains longitude.
+        Each row is a geographic point
+    coordinates : 2d array
+        First column contains latitude, second column contains longitude. Each
+        row is a geographic point
+    threshold : float
+        distance threshold in km over which no neighbor will be found. Those
+        are assigned with a -1 index
+
+    Returns
+    -------
+    np.array
+        with so many rows as coordinates containing the centroids indexes
+    """
+    # Construct tree from centroids
+    tree = sp.spatial.KDTree(np.radians(centroids))
+    # Select unique exposures coordinates
+    _, idx, inv = np.unique(coordinates, axis=0, return_index=True,
+                            return_inverse=True)
+
+    # query the k closest points of the n_points using dual tree
+    dist, assigned = tree.query(np.radians(coordinates[idx]), k=1, p=2, workers=-1)
+
+    # Raise a warning if the minimum distance is greater than the
+    # threshold and set an unvalid index -1
```

Comment:
Ok, I see we've been doing this also for the other two methods; but is setting an index of -1 always save? I.e. in no circumstance it will just silently take the last index in the array as a result, if I use this afterwards?

Reply:
It's definitely not save, but there is no alternative.

Reply:
Well, a Nan could also have been used, but -1 is probably still better.

Reply:
Not if the data type is `int`. For `NaN` we need to switch the data type to `float`.

Reply:
Sure, but that could be worth it in principle. But probably not in the current state.